### PR TITLE
Translate Context

### PR DESCRIPTION
**Pull request by ricardoerl**
Spanish translation of Context documentation.

Todo:
- [x] When to Use Context
- [x] Before You Use Context
- [x] API
  - [x] React.createContext
  - [x] Context.Provider
  - [x] Class.contextType
  - [x] Context.Consumer
- [x] Examples
  - [x] Dynamic Context
  - [x] Updating Context from a Nested Component
  - [x] Consuming Multiple Contexts
- [x] Caveats
- [x] Legacy API

### CODE REVIEW

**Comment by alejandronanez** [comment]
Hey @ricardoerl, thanks for contributing!

Let us know if you need any help with this!

**Comment by alejandronanez** [comment]
Hey @ricardoerl I’ll take a look tonight, thanks!

File: content/docs/context.md [hunk 1]
```diff
@@ -155,27 +156,27 @@ class MyClass extends React.Component {
...
+A la propiedad `contextType` en una clase se le puede asignar un objeto Context creado por [`React.createContext()`](#reactcreatecontext). Esto te permite consumir el valor actual más cercano de ese Context utilizando `this.context`. Puedes hacer referencia a esto en cualquiera de los métodos del ciclo de vida, incluida la función de renderizado.
 
 > Note:
 >
-> You can only subscribe to a single context using this API. If you need to read more than one see [Consuming Multiple Contexts](#consuming-multiple-contexts).
+> Solo puedes suscribirte a un solo Context usando esta API. Si necesitas leer más de una, lee [Consuming Multiple Contexts](#consuming-multiple-contexts).
 >
-> If you are using the experimental [public class fields syntax](https://babeljs.io/docs/plugins/transform-class-properties/), you can use a **static** class field to initialize your `contextType`.
+> Si está utilizando la [sintaxis experimental de campos de clase pública](https://babeljs.io/docs/plugins/transform-class-properties/), puedes usar un campo de clase **static** para inicializar tu `contextType`.
```
Comment (Darking360) [inline content/docs/context.md]:
```suggestion
> Si estás utilizando la [sintaxis experimental de campos de clase pública](https://babeljs.io/docs/plugins/transform-class-properties/), puedes usar un campo de clase **static** para inicializar tu `contextType`.
```

File: examples/context/theme-detailed-app.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import {ThemeContext, themes} from './theme-context';
 import ThemedButton from './themed-button';
 
-// An intermediate component that uses the ThemedButton
+// Un componente intermedio que utiliza ThemedButton.
 function Toolbar(props) {
   return (
     <ThemedButton onClick={props.changeTheme}>
-      Change Theme
+      Cambiar tema
```
Comment (Darking360) [inline examples/context/theme-detailed-app.js]: Solo estamos traduciendo los comentarios 🥇 

```suggestion
      Change Theme
```

File: content/docs/context.md [hunk 3]
```diff
@@ -211,9 +212,9 @@ A more complex example with dynamic values for the theme:
 **app.js**
 `embed:context/theme-detailed-app.js`
 
-### Updating Context from a Nested Component {#updating-context-from-a-nested-component}
+### Actualización Context de un componente anidado {#updating-context-from-a-nested-component}
```
Comment (Darking360) [inline content/docs/context.md]:
```suggestion
### Actualizando Context desde un componente anidado {#updating-context-from-a-nested-component}
```

File: examples/context/reference-caveats-problem.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ class App extends React.Component {
   render() {
     // highlight-range{2}
     return (
-      <Provider value={{something: 'something'}}>
+      <Provider value={{something: 'algo'}}>
```
Comment (Darking360) [inline examples/context/reference-caveats-problem.js]:
```suggestion
      <Provider value={{something: 'something'}}>
```

File: examples/context/reference-caveats-solution.js [hunk 5]
```diff
@@ -3,7 +3,7 @@ class App extends React.Component {
     super(props);
     // highlight-range{2}
     this.state = {
-      value: {something: 'something'},
+      value: {something: 'algo'},
```
Comment (Darking360) [inline examples/context/reference-caveats-solution.js]:
```suggestion
      value: {something: 'something'},
```

File: content/docs/context.md [hunk 6]
```diff
@@ -224,28 +225,28 @@ It is often necessary to update the context from a component that is nested some
...
 `embed:context/multiple-contexts.js`
 
-If two or more context values are often used together, you might want to consider creating your own render prop component that provides both.
+Si dos o más valores de contexto se usan a menudo juntos, es posible que desees considerar la creación de tu propio componente de procesamiento que proporcione ambos.
 
-## Caveats {#caveats}
+## Advertencias {#caveats}
 
-Because context uses reference identity to determine when to re-render, there are some gotchas that could trigger unintentional renders in consumers when a provider's parent re-renders. For example, the code below will re-render all consumers every time the Provider re-renders because a new object is always created for `value`:
+Debido a que `Context` usa la identidad de referencia para determinar cuándo se debe volver a renderizar, hay algunos errores que podrían provocar renderizados involuntarios en los consumidores cuando se vuelve a renderizar en el padre del proveedor. Por ejemplo, el código a continuación volverá a representar a todos los consumidores cada vez que el Proveedor vuelva a renderizar porque siempre se crea un nuevo objeto para `value`:
```
Comment (Darking360) [inline content/docs/context.md]:
```suggestion
Debido a que `Context` usa la identidad por referencia para determinar cuándo se debe volver a renderizar, hay algunos errores que podrían provocar renderizados involuntarios en los consumidores cuando se vuelve a renderizar en el padre del proveedor. Por ejemplo, el código a continuación volverá a renderizar a todos los consumidores cada vez que el `Proveedor` se vuelva a renderizar porque siempre se crea un nuevo objeto para `value`:
```

File: content/docs/context.md [hunk 7]
```diff
@@ -224,28 +225,28 @@ It is often necessary to update the context from a component that is nested some
...
 `embed:context/reference-caveats-solution.js`
 
-## Legacy API {#legacy-api}
+## API antigua {#legacy-api}
 
-> Note
+> Nota
 > 
-> React previously shipped with an experimental context API. The old API will be supported in all 16.x releases, but applications using it should migrate to the new version. The legacy API will be removed in a future major React version. Read the [legacy context docs here](/docs/legacy-context.html).
+> React previamente liberado con el API experimental de Context. La antigua API será compatible con todas las versiones 16.x, pero las aplicaciones que la utilicen deberían migrar a la nueva versión. La API antigua se eliminará en una importante futura versión de React. Leer la [documentación antigua de Context aquí](/docs/legacy-context.html).
```
Comment (Darking360) [inline content/docs/context.md]:
```suggestion
> React previamente había liberado el API experimental de Context. La antigua API será compatible con todas las versiones 16.x, pero las aplicaciones que la utilicen deberían migrar a la nueva versión. La API antigua se eliminará en una futura versión importante de React. Lee la [documentación antigua de Context aquí](/docs/legacy-context.html).
```

**Comment by ricardoerl** [comment]
@Darking360 changes approved. Thanks a lot for the help 🙌

**Comment by carburo** [comment]
The build is failing because you have long line comments in .js files. You should run Prettier.

**Comment by Darking360** [comment]
@ricardoerl Greetings! There are some changes left, maybe you didn't expand the selection of feedback, it has a `Load more` or `Show other N conversations` button. Please add those and then we'd be good to go 💪

**Pull request merged**